expression/builtin_time.go

```diff
@@ -264,6 +264,15 @@ func convertTimeToMysqlTime(t time.Time, fsp int8, roundMode types.RoundMode) (t
 	return types.NewTime(types.FromGoTime(tr), mysql.TypeDatetime, fsp), nil
 }
 
+func getDateAddOrSubReturnTypeByUnit(dateType uint8, unit string) uint8 {
+	isClockUnit := types.IsClockUnit(unit)
+	dateTp := mysql.TypeDate
```

Comment:
Better to not use a new temp vairable, you can return the type like this
```
if dateType == mysql.TypeDatetime || isClockUnit {
    return mysql.TypeDatetime
}
return mysql.TypeDate
```

Reply:
done